runtime/doc/lsp.txt

```diff
@@ -0,0 +1,359 @@
+*lsp.txt* The Language Server Protocol
+
+        NVIM REFERENCE MANUAL
+
+
+Neovim Language Server Protocol (LSP) API
+
+Neovim exposes a powerful API that conforms to Microsoft's published Language
+Server Protocol specification. The documentation can be found here:
+
+	https://microsoft.github.io/language-server-protocol/
+
+
+================================================================================
+                                                            *lsp-server-config*
+
+Neovim only exposes a powerful API for the language server protocol. To get
+the real benefits of this API, a language server must be installed. Many
+examples can be found here:
+
+	https://microsoft.github.io/language-server-protocol/implementors/servers/
+
+After installing a language server to your machine, you must let Neovim know
+how to start and interact with that language server. To do so, use the
+following APIs:
+
+                                                      *lsp#add_server_config()*
+lsp#add_server_config({config}):
+	Used to add a language server.
+
+	This will add an autocmd that will start the server and handle any other
+	startup activities whenever the |FileType| event is triggered for {ftype}.
+
+	Parameters: ~
+		{config} (dictionary): 
+			`filetype` (string|array):
+				A string or list of strings of filetypes to
+				associate with this server.
+
+			`server_name` (string) (optional):
+				You can pass any name as server_name.
+				If you want to use multiple language servers
+				for one filetype, be sure to add server_name.
+
+			`cmd` (dictionary): Executing language server command.
```

Comment:
This is needlessly different from the `jobstart()` syntax. `cmd` should accept a single array where first item is the executable and the rest are args (that luv uses a different syntax internally is irrelevant to the user).